modules/django-drive/django_drive/drive/viewsets.py

```diff
@@ -0,0 +1,152 @@
+import apiclient
+from demo import settings
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from googleapiclient import discovery
+from google.oauth2.credentials import Credentials
+
+from .serializers import (ShareFileSerializer, CreateFolderSerializer, UploadFileSerializer)
+
+
+def init_service(token):
+    """
+    The method authorizes a user and creates an access point to interact with Google Drive.
+    Gives access to the user to access the files and folders of the drive
+    """
+    creds = Credentials(token=token)
+    return discovery.build('drive', 'v3', credentials=creds)
+
+
+class GetDriveFilesViewSet(APIView):
+    """
+    This class without any parameters return all the files and folders from the Google Drive. 
+    Filters results according to the provided queries.
+    """
+
+    def get(self, request, *args, **kwargs):
+        """
+        :header:
+            Authorization: send access token in header without keyword 'Bearer'
+        :query_param str query: To search for a specific set of files or folders, use the query. 
+        :param num pageSize: The maximum number of files to return per page
+        :param str pageToken: The token for continuing a previous list request on the next page. This should be set to the value of 'nextPageToken' from the previous response.
+        :param str mimeType: Specifies if the user wants to get files or folders supported values are `application/vnd.google-apps.file` and `application/vnd.google-apps.folder`
+        :return: Returns list of the files and folder from user's Google Drive.
+        """
+        try:
+            drive_service = init_service(token=request.META.get('HTTP_AUTHORIZATION'))
```

Comment:
is it possible to move this token to ENV Variables? how often does it change ?

Reply:
> is it possible to move this token to ENV Variables? how often does it change ?

@shahraizali This oAuth2 access token with scope to access the drive APIs and It expires every 1-2 hours.
Basically, when we request to get an access token it comes with three things access_token, refresh_token and expires_in. So we need to check expires_in time and before expiration, we have to hit API with refresh_token to get a new token. So It can't be static or remain the same for a long time.